dataplane/internal/engine/engine.go

```diff
@@ -233,45 +282,249 @@ func (e *Engine) AddLayer3PuntRule(ctx context.Context, portName string, ip []by
 	return nil
 }
 
-// AddIPRoute adds a route to the FIB with the input next hops.
-func (e *Engine) AddIPRoute(ctx context.Context, v4 bool, ip, mask []byte, vrf uint64, nextHops []*dpb.NextHop) error {
-	fib := fibV6Table
-	if v4 {
-		fib = fibV4Table
+// prefixToPrimitives returns the primitive types of the route prefix.
+// ip addr bytes, ip mask bytes, is ipv4, vrf id, error.
+func prefixToPrimitives(prefix *dpb.RoutePrefix) ([]byte, []byte, bool, uint64, error) {
+	var ip []byte
+	var mask []byte
+	var isIPv4 bool
+	vrf := prefix.GetVrfId()
+
+	switch pre := prefix.GetPrefix().(type) {
+	case *dpb.RoutePrefix_Str:
+		_, ipNet, err := net.ParseCIDR(pre.Str)
```

Comment:
(no action required) I recommend checking out netip package, it was created because net was a bit hard to use.

Reply:
it's easier to get the mask bytes with net package, but yeah otherwise netip is nice